### PR TITLE
v1.20.0 upgrade guide

### LINKED ISSUE #107: Upgrade guide for the Vault operator
We need a place in the documentation for the vault operator at this time, but we will probably need it for other components as well.

- [x] Propose a structure for upgrade guides in the documentation @fekete-robert 
- [x] Review docs PR @ramizpolic 
- [x] Test and document upgrading the vault operator from 1.19 to 1.20 @akijakya 
- [ ] Update Vault Operator upgrade guide in the docs page

### PR DESCRIPTION
**Pull request by akijakya**
## Overview

Fixes #107

### CODE REVIEW

File: content/docs/operator/upgrade.md [hunk 1]
```diff
@@ -10,3 +10,21 @@ To upgrade the Vault operator, complete the following steps.
 1. Check the [release notes of Vault](https://developer.hashicorp.com/vault/docs/release-notes) for any special upgrade instructions. Usually there are no instructions, but it's better to be safe than sorry.
 1. Adjust the [spec.image field in the Vault custom resource](https://github.com/bank-vaults/vault-operator/blob/v{{< param "latest_operator_version" >}}/deploy/examples/cr.yaml#L7). If you are using the Vault Helm chart, adjust the [image.tag field in the values.yaml](https://github.com/bank-vaults/vault-helm-chart/blob/v{{< param "latest_version" >}}/vault/values.yaml#L13).
 1. The Bank-Vaults operator (or the Vault Helm chart) updates the StatefulSet. (It does not take the HA leader into account in HA scenarios, but this has never caused any issues so far.)
+
+### v1.20.0 upgrade guide
```
Comment (fekete-robert) [inline content/docs/operator/upgrade.md:14]:
```suggestion
## v1.20.0 upgrade guide
```

**Review by fekete-robert — APPROVED** [review]

File: content/docs/operator/upgrade.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
+title: Upgrade Vault operator
```
Comment (ramizpolic) [inline content/docs/operator/upgrade.md:2]:
```suggestion
title: Upgrade strategies
```

File: content/docs/operator/upgrade.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+weight: 50
+aliases:
+- /docs/operator/upgrade-vault/
+---
+
+To upgrade the Vault operator, complete the following steps.
```
Comment (ramizpolic) [inline content/docs/operator/upgrade.md:8]:
```suggestion
## Upgrade Vault

To upgrade the Vault, complete the following steps.
```

File: content/docs/operator/upgrade.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+1. Check the [release notes of Vault](https://developer.hashicorp.com/vault/docs/release-notes) for any special upgrade instructions. Usually there are no instructions, but it's better to be safe than sorry.
+1. Adjust the [spec.image field in the Vault custom resource](https://github.com/bank-vaults/vault-operator/blob/v{{< param "latest_operator_version" >}}/deploy/examples/cr.yaml#L7). If you are using the Vault Helm chart, adjust the [image.tag field in the values.yaml](https://github.com/bank-vaults/vault-helm-chart/blob/v{{< param "latest_version" >}}/vault/values.yaml#L13).
+1. The Bank-Vaults operator (or the Vault Helm chart) updates the StatefulSet. (It does not take the HA leader into account in HA scenarios, but this has never caused any issues so far.)
```
Comment (ramizpolic) [inline content/docs/operator/upgrade.md:12]:
```suggestion
1. The Vault Helm chart updates the StatefulSet. It does not take the HA leader into account in HA scenarios, but this has never caused any issues so far.
```

File: content/docs/operator/upgrade.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+## v1.20.0 upgrade guide
```
Comment (ramizpolic) [inline content/docs/operator/upgrade.md:14]:
```suggestion
## Upgrade Vault operator

### v1.20.0 upgrade guide
```

**Review by ramizpolic — CHANGES_REQUESTED** [review]
Sorry for the confusion with the upgrade strategies. I have added items to address two scenarios: upgrading Vault, and upgrading Operator.

**Comment by ramizpolic** [comment]
We can merge once these items are resolved and DCO passing.

**Review by ramizpolic — APPROVED** [review]

**Pull request merged**